### DISCUSSION TITLE
How to hide certain columns on the initial load of the app?

### DISCUSSION THREAD

**Discussion by sweezy793** [ghiscoding/Angular-Slickgrid#834] [Q&A]
Hi, my implementation is on the start of my app certain columns should stay hidden and get enabled only when selected from headerMenu or gridMenu. I tried with presets, but with presets those columns get hidden while the app is loading, but after data gets fetched those columns are visible.

Can anyone help me with this?

**Comment by ghiscoding** [comment]
I would guess that you mean Column Picker (not Header Menu), the Grid Presets are the way to go, it has a `columns` property which will end up being the visible columns (whichever columns isn't part of that property won't show up). For example I did that in some projects to add all columns including `id` but I didn't want to show that column unless chosen by the pickers, so I use the Grid Presets and use `filter` on the definitions array and remove the `id` column then pass this new array to the `columns` property and that works. The presets shouldn't be affected by the data, unless you've end up changing the column definitions after the grid options? Make sure your grid options are defined after the column definitions, not the inverse. If you want to change the list of visible columns after the first page load, you can use the method `changeColumnsArrangement` from the Grid State Service. 

Please ask questions on Stack Overflow as much as possible, many more users can answer questions on that platform. 
Thanks ⭐
